client/src/components/TaskReflectionModal/TaskReflectionModal.js

```diff
@@ -39,6 +39,20 @@ const TaskReflectionModal = ({
     onDone(reflectionComments, chosenDot + 1);
   };
 
+  useEffect(() => {
+    if (readOnly) {
+      setErrorMessage("");
+      getTaskReflection()
```

Comment:
We can restructure this part of the component where the page itself will pass in the all of the reflection data through props (i.e. the comment, chosen dots, etc.). This will reduce the need for having the `.then` clause.

Reply:
With this restructure, it might even remove the need for the `useEffect` hook

Reply:
Restructured to make all fetches together with `getTasks`.